### DISCUSSION TITLE
Workflow Rollback

### DISCUSSION THREAD

**Discussion by JHancox** [Project-MONAI/monai-deploy-workflow-manager#42] [Ideas]
Have we considered rolling things back as part of the workflow functionality? Let's say half of a workflow executes and then someone realises that there is a problem with the study acquisition. Not only do we need to provide a mechanism to stop the workflow, but we may need to undo outputs from previous steps (e.g. an interim step may have committed data to PACS/EHR.). This would then need to recorded in the logs etc too.

**Comment by dbericat** [comment]
Good question. Important to clarify first whether this functionality should be part of MONAI or outside in a different system as part of the governance of the final solution. i.e. AIDE.
